### PR TITLE
Skiko events update.

### PR DESCRIPTION
**Pull request by Rsedaikin**

### CODE REVIEW

**Comment by olonho** [comment]
CI fails.

File: skiko/src/jsMain/kotlin/org/jetbrains/skiko/SkiaLayer.js.kt [hunk 1]
```diff
@@ -66,36 +49,26 @@ actual open class SkiaLayer {
...
-            var mask = SkikoMouseButtons.NONE
-            skikoView?.onPointerEvent(translatePointerEvent(
-                event, true, SkikoPointerEventKind.UP))
+            skikoView?.onPointerEvent(toSkikoEvent(event, true, SkikoPointerEventKind.UP))
         })
         htmlCanvas.addEventListener("pointermove", { event ->
             event as MouseEvent
-            skikoView?.onPointerEvent(translatePointerEvent(
-                event, false, SkikoPointerEventKind.MOVE))
+            skikoView?.onPointerEvent(toSkikoEvent(event, false, SkikoPointerEventKind.MOVE))
```
Comment (eymar) [inline skiko/src/jsMain/kotlin/org/jetbrains/skiko/SkiaLayer.js.kt:60]:
```
toSkikoEvent(event, false, SkikoPointerEventKind.MOVE)
```
as I understand `false` here means that there are no buttons pressed.

But it's possible that pointer can be moved while some buttons are pressed, so users will receive skiko move event without any information about pressed buttons. Do I understand current logic correctly?

I see `false` was there before these changes, so I'm wondering if `false` is intentional?

File: samples/SkiaJvmSample/build.gradle.kts [hunk 2]
```diff
@@ -76,6 +76,11 @@ tasks.register("runWithTransparency") {
     dependsOn(casualRun)
 }
 
+tasks.register("runInterop") {
+    additionalArguments += mapOf("skiko.interop" to "true")
```
Comment (olonho) [inline samples/SkiaJvmSample/build.gradle.kts:80]: Better smth like `skiko.swing.interop`.

File: samples/SkiaJvmSample/src/main/kotlin/SkiaJvmSample/App.kt [hunk 3]
```diff
@@ -14,7 +14,10 @@ import javax.imageio.ImageIO
 fun main() {
     val windows = 1
```
Comment (olonho) [inline samples/SkiaJvmSample/src/main/kotlin/SkiaJvmSample/App.kt:15]: Shalln't we make `windows` property as well?

File: samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt [hunk 4]
```diff
@@ -94,12 +89,23 @@ class Clocks: SkikoView {
...
+            SkikoPointerEventKind.MOVE -> {
+                xpos = event.x
+                ypos = event.y
+            }
+            else -> { println("PointerEvent: ${event.modifiers} + ${event.kind} + ${event.buttons}") }
         }
     }
 
+    override fun onInputEvent(event: SkikoInputEvent) {
+        // TODO: provide examplec
```
Comment (olonho) [inline samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt:102]: example

File: samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt [hunk 5]
```diff
@@ -94,12 +89,23 @@ class Clocks: SkikoView {
...
+    }
+
+    override fun onKeyboardEvent(event: SkikoKeyboardEvent) {
+        println("KeyboardEvent: ${event.kind} ${event.modifiers} + ${event.code}")
```
Comment (olonho) [inline samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt:106]: unneeded print

File: skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt [hunk 6]
```diff
@@ -1,11 +1,97 @@
...
+                add("BUTTON_8")
+            }
+            if (isEmpty()) {
+                add("NONE")
+            }
+        }.toString()
+    }
+}
+
+class SkikoInputModifiers(val value: Int) {
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt:60]: inline class

File: skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt [hunk 7]
```diff
@@ -1,11 +1,97 @@
 package org.jetbrains.skiko
 
 // Clicked mouse buttons bitmask.
-object SkikoMouseButtons {
-    const val NONE = 0
-    const val LEFT = 1 shl 0
-    const val RIGHT = 1 shl 1
-    const val MIDDLE = 1 shl 2
+class SkikoMouseButtons(val value: Int) {
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt:4]: inline class

File: skiko/src/macosMain/kotlin/org/jetbrains/skiko/Convertors.macos.kt [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+): SkikoKeyboardEvent {
+    return SkikoKeyboardEvent(
+        event.keyCode.toInt(),
+        toSkikoModifiers(event),
+        kind,
+        event
+    )
+}
+
+fun toSkikoMouseButtons(event: NSEvent): SkikoMouseButtons {
```
Comment (olonho) [inline skiko/src/macosMain/kotlin/org/jetbrains/skiko/Convertors.macos.kt:40]: Make them internal or private.

File: skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt [hunk 9]
```diff
@@ -1,11 +1,97 @@
...
+            if (has(SkikoInputModifiers.CONTROL)) {
+                add("CONTROL")
+            }
+            if (has(SkikoInputModifiers.ALT)) {
+                add("ALT")
+            }
+            if (has(SkikoInputModifiers.SHIFT)) {
+                add("SHIFT")
+            }
+            if (isEmpty()) {
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt:90]: Seems wrong, at least for me gives `PointerEvent: [EMPTY] + DOWN + [LEFT]`

File: skiko/src/macosMain/kotlin/org/jetbrains/skiko/Convertors.macos.kt [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    var result = 0
+    val mask = event.buttonMask.toInt()
+    if ((mask and 1) != 0 || event.buttonNumber == 0L) {
+        result = result.or(SkikoMouseButtons.LEFT.value)
+    }
+    if ((mask and 2) != 0 || event.buttonNumber == 1L) {
```
Comment (olonho) [inline skiko/src/macosMain/kotlin/org/jetbrains/skiko/Convertors.macos.kt:46]: Shalln't we process other buttons? I have a multi-button mouse and see the following NSEvent when pressing other buttons, if add smth like
```
override fun otherMouseDown(event: NSEvent) {
                skikoView?.onPointerEvent(toSkikoEvent(event, SkikoPointerEventKind.DOWN, nsView))
            }
            override fun otherMouseUp(event: NSEvent) {
                skikoView?.onPointerEvent(toSkikoEvent(event, SkikoPointerEventKind.UP, nsView))
            }
```

```
e=NSEvent: type=OtherMouseDown loc=(170,264) time=13656.9 flags=0 win=0x7feec27095f0 winNum=540 ctxt=0x0 evNum=3 click=1 buttonNumber=2 pressure=1 deviceID:0x0 subtype=0
PointerEvent: [EMPTY] + DOWN + [NONE]
e=NSEvent: type=OtherMouseUp loc=(170,264) time=13657.1 flags=0 win=0x7feec27095f0 winNum=540 ctxt=0x0 evNum=3 click=1 buttonNumber=2 pressure=0 deviceID:0x0 subtype=0
PointerEvent: [EMPTY] + UP + [NONE]
```
for pressing mouse wheel, and no events for other buttons on MX Master 3.

File: samples/SkiaJvmSample/build.gradle.kts (hunk 2 above)
Comment (Rsedaikin) [reply samples/SkiaJvmSample/build.gradle.kts:80]: Done.

File: samples/SkiaJvmSample/src/main/kotlin/SkiaJvmSample/App.kt (hunk 3 above)
Comment (Rsedaikin) [reply samples/SkiaJvmSample/src/main/kotlin/SkiaJvmSample/App.kt:15]: I made via [main(args: Array<String>)]

File: samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt (hunk 4 above)
Comment (Rsedaikin) [reply samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt:102]: Done.

File: samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt (hunk 5 above)
Comment (Rsedaikin) [reply samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt:106]: I've removed all event output for now - I have thoughts on a better future example.

File: skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt (hunk 6 above)
Comment (Rsedaikin) [reply skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt:60]: Done.

File: skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt (hunk 7 above)
Comment (Rsedaikin) [reply skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt:4]: Done.

File: skiko/src/macosMain/kotlin/org/jetbrains/skiko/Convertors.macos.kt (hunk 8 above)
Comment (Rsedaikin) [reply skiko/src/macosMain/kotlin/org/jetbrains/skiko/Convertors.macos.kt:40]: Done.

File: skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt (hunk 9 above)
Comment (Rsedaikin) [reply skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt:90]: I've refactored [toString] method to be cleaner.

File: samples/SkiaJvmSample/src/main/kotlin/SkiaJvmSample/App.kt [hunk 11]
```diff
@@ -120,3 +116,17 @@ fun createWindow(title: String, exitOnClose: Boolean) = SwingUtilities.invokeLat
     skiaLayer.paint(window.graphics)
     window.isVisible = true
 }
+
+private fun parseArgs(args: Array<String>): Int {
```
Comment (olonho) [inline samples/SkiaJvmSample/src/main/kotlin/SkiaJvmSample/App.kt:120]: Why not property as everything else?

Comment (Rsedaikin) [reply samples/SkiaJvmSample/src/main/kotlin/SkiaJvmSample/App.kt:120]: I was thinking about passing the number of windows parameter via the command line, something like `./gradlew <run task> <window count>` so I had 2 options:
 - `./gradlew run -Dskiko.example.window.count=2`
 - `./gradlew run --args='2'`
 the last one is much shorter.

**Pull request merged**